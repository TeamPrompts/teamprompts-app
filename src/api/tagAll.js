export const ALL = 'all';
const tagAll = { name: ALL, slug: ALL };

export default tagAll;
