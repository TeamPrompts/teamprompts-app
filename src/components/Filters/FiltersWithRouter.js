import React from 'react';
import {
  HashRouter as Router,
  Route,
  Switch,
  useParams
} from 'react-router-dom';
import Filters from './Filters';

function FiltersWithRouter({ filter, fitbs, onClick, tags }) {
  function Child() {
    const { id } = useParams();
    const tag = tags.find(tag => tag.slug === id);
    if (tag && filter !== tag) {
      onClick(tag);
    }
    return (
      <Filters filter={filter} fitbs={fitbs} onClick={onClick} tags={tags} />
    );
  }

  return (
    <Router>
      <Switch>
        <Route path="/:id" children={<Child />} />
        <Route path="/" children={<Child />} />
      </Switch>
    </Router>
  );
}

export default FiltersWithRouter;
