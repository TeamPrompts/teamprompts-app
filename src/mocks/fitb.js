import { SEPARATOR } from '../constants';

export const SOURCE = `Do you know the way to ${SEPARATOR}?`;

const fitb = {
  examples: ['San Jose'],
  id: 'receM4Gus7HH43oLM',
  prompts: ['[place]'],
  source: SOURCE,
  tags: ['recTqD26ElVvXKVad']
};

export default fitb;
