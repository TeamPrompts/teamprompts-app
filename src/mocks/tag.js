import tags from './tags';

const tag = tags.find((tag) => tag.id === 'recTqD26ElVvXKVad');

export default tag;
