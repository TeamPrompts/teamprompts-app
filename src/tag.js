const tag = {
  fitbs: [
    'receM4Gus7HH43oLM',
    'rec99C9lxMl6aSedq',
    'rec590aP1VRIE8f23',
    'recNsahDzKCx0IQPk'
  ],
  id: 'recTqD26ElVvXKVad',
  name: 'retrospectives'
};

export default tag; // INFO: this is a mock
