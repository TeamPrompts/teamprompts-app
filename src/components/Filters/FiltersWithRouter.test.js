import { render, fireEvent } from '@testing-library/react';
import React from 'react';
import { create } from 'react-test-renderer';
import fitb from '../../mocks/fitb';
import tag from '../../mocks/tag';
import tagAll from '../../api/tagAll';
import FiltersWithRouter from './FiltersWithRouter';

describe('FiltersWithRouter', () => {
  const mockOnClick = jest.fn();
  const props = {
    filter: tagAll,
    fitbs: [fitb, { ...fitb, examples: ['Los Angeles'] }],
    onClick: mockOnClick,
    tags: [tag]
  };

  it('to match snapshot', () => {
    const tree = create(<FiltersWithRouter {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('click', () => {
    const { getByTestId } = render(<FiltersWithRouter {...props} />);

    fireEvent.click(getByTestId(tag.name));

    expect(mockOnClick).toBeCalled();
  });
});
