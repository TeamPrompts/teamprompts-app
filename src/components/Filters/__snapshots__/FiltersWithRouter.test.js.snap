// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FiltersWithRouter to match snapshot 1`] = `
<ul
  className="border-b-2 flex flex-wrap"
>
  <li
    className="mx-1 my-1"
  >
    <a
      href="#/all"
      onClick={[Function]}
    >
      <button
        className="focus:outline-none hover:bg-blue-700 px-4 rounded-full text-white bg-blue-900"
        data-testid="all"
        onClick={[Function]}
        type="button"
      >
        all (2)
      </button>
    </a>
  </li>
  <li
    className="mx-1 my-1"
  >
    <a
      href="#/retrospectives"
      onClick={[Function]}
    >
      <button
        className="focus:outline-none hover:bg-blue-700 px-4 rounded-full text-white bg-blue-500"
        data-testid="retrospectives"
        onClick={[Function]}
        type="button"
      >
        retrospectives (2)
      </button>
    </a>
  </li>
</ul>
`;
